modules/tensor_mechanics/src/materials/FiniteStrainCrystalPlasticity.C

```diff
@@ -1092,3 +1111,32 @@ FiniteStrainCrystalPlasticity::elasticTangentModuli()
 {
   return _elasticity_tensor[_qp];//update jacobian_mult
 }
+
+bool
+FiniteStrainCrystalPlasticity::line_search_update(const Real rnorm_prev, const RankTwoTensor dpk2)
+{
+  Real rnorm;
+  RankTwoTensor resid;
+  bool iter_flag = true;
+  Real step = 1.0;
+
+  while (iter_flag)
+  {
+    iter_flag = false;
+
+    _pk2_tmp = _pk2_tmp - step * dpk2;
+    step /= 2.0;
+    _pk2_tmp = _pk2_tmp + step * dpk2;
+
+    calcResidual(resid);
+    rnorm = resid.L2norm();
+
+    if (rnorm > rnorm_prev && step > _min_lsrch_step)
+      iter_flag = true;
```

Comment:
This is not elegant. The ```iter_flag``` variable is not necessary. A ```do { } while()``` construct would fit exactly here.